src/3d/qgs3dmapscene.cpp

```diff
@@ -247,6 +247,36 @@ void Qgs3DMapScene::viewZoomFull()
   mCameraController->resetView( 1.5 * std::sqrt( a * a - side * side ) );  // assuming FOV being 45 degrees
 }
 
+void Qgs3DMapScene::viewExtent( const QgsRectangle &extent )
+{
+  QgsPointXY center = extent.center();
+  QgsVector3D centerWrld = mMap.mapToWorldCoordinates( QVector3D( center.x(), center.y(), 0 ) );
+  QgsVector3D p1 = mMap.mapToWorldCoordinates( QVector3D( extent.xMinimum(), extent.yMinimum(), 0 ) );
+  QgsVector3D p2 = mMap.mapToWorldCoordinates( QVector3D( extent.xMaximum(), extent.yMaximum(), 0 ) );
+
+  float side = std::max( std::abs( p1.x() - p2.x() ), std::abs( p1.z() - p2.z() ) );
```

Comment:
I wonder if average of the two wouldn't be better approximation, especially if the difference between aspect ratio of 2D and 3D is higher